sdk/core/Azure.Core/src/Pipeline/Policies/BearerTokenAuthenticationPolicy.cs

```diff
@@ -50,15 +50,14 @@ public override void Process(HttpPipelineMessage message, ReadOnlyMemory<HttpPip
 
         public async Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline, bool async)
         {
-            string token = async ?
-                    await _credential.GetTokenAsync(_scopes, message.CancellationToken).ConfigureAwait(false) :
-                    _credential.GetToken(_scopes, message.CancellationToken);
-
-            if (token != _currentToken)
+            if (DateTimeOffset.UtcNow >= _refreshOn)
```

Comment:
Should we consider a fuzzy window here to account for potential clock skew?   Something like "If this expires Now +- 5 minutes" or similar?

Reply:
_refreshOn is already 2 minutes ahead of the actual expiration.  Do you think this is insufficient?

~~~ c#
                _refreshOn = token.ExpiresOn - TimeSpan.FromMinutes(2);
~~~

Reply:
Sorry, missed that.   We had seen clock skew up to 5 minutes between VM instances in the same data center for machines of the same configuration when hosting on 1000+ instances, despite running an NTP sync once a day.  I'm not sure that I'd call that the norm, but it's possible.   Were it me, I'd probably give it a 5 minute cushion but 2 should be good for the majority.

Reply:
I'll merge this as is for now since I didn't change the buffer time in this PR.  I've created issue #6528 to track this.